### PR TITLE
0.18.x xrefs, some rewrites to "Architecture"

### PR DESCRIPTION
**Pull request by StackScribe**
Signed-off-by: Meg McRoberts <meg.mcroberts@dynatrace.com>

* Fixed xrefs in "Concepts" articles to point to 0.18.x
* Did a few little rewrites to Architecture -- still not perfect but incremental improvements that should be critiqued carefully

### CODE REVIEW

File: content/docs/concepts/architecture/index.md [hunk 1]
```diff
@@ -9,73 +9,137 @@ Keptn is an event-based control plane for continuous delivery and automated oper
...
-**Note:** A dedicated Keptn API section is provided [here](../../0.15.x/reference/api/), which helps you to access the API and to explore the available endpoints.
+The documentation shows you how to access the API and to explore the available endpoints.
 
 ### mongodb-service
 
-The *mongodb-datastore* provides means to store event data in a MongoDB deployed in your Keptn namespace. Hence, the service provides the REST endpoint `/events` to query events.
+The *mongodb-datastore* stores event data in a MongoDB deployed in your Keptn namespace.
+Hence, the service provides the REST endpoint `/events` to query events.
 
 ### configuration-service
```
Comment (thschue) [inline content/docs/concepts/architecture/index.md:69]: As far as I know, the configuration service does not exist anymore in newer keptn versions and the successor is the resource service

File: content/docs/concepts/architecture/index.md [hunk 2]
```diff
@@ -9,73 +9,137 @@ Keptn is an event-based control plane for continuous delivery and automated oper
...
+
+Keptn-services react to `.triggered` Keptn events that are sent by the shipyard controller.
+They perform continuous delivery tasks like deploying or promoting a service
+and orchestrational tasks for automating operations.
+Those services can be easily plugged into a task sequence
+to extend the delivery pipeline or to further automate operations.
+Execution plane services subscribe to events via a
+[distributor](../../0.18.x/reference/miscellaneous/distributor) sidecar
+that forwards incoming `.triggered` events to them.
+These distributor sidecars can also be used to send `.started` and `.finished` events
```
Comment (thschue) [inline content/docs/concepts/architecture/index.md:105]: I think in the meanwhile, two options for that exist, the distributor sidecar but there's also a control plane connector library.

**Review by thschue — CHANGES_REQUESTED** [review]

File: content/docs/concepts/architecture/index.md [hunk 3]
```diff
@@ -9,73 +9,137 @@ Keptn is an event-based control plane for continuous delivery and automated oper
...
+It must be [installed](../../install/cli-install)
+on the local machine and is used to send commands to Keptn.
+To communicate with Keptn, you need to know a shared secret
+that is managed by the [Secret Service](../secrets).
+The shared secret is generated during the installation and verified by the *api* component.
 
 ## Keptn Bridge
 
-The *Keptn Bridge* provides a user interface that shows all Keptn-managed projects and services. Besides, it gives an overview of the staging environment. 
+The *Keptn Bridge* provides a user interface that shows all projects and services that Keptn manages
```
[the remark this answers is not in the document]
Comment (StackScribe) [reply content/docs/concepts/architecture/index.md:32]: done

File: content/docs/concepts/architecture/index.md [hunk 4]
```diff
@@ -9,73 +9,137 @@ Keptn is an event-based control plane for continuous delivery and automated oper
...
+and gives an overview of the staging environment.
```
[the remark this answers is not in the document]
Comment (StackScribe) [reply content/docs/concepts/architecture/index.md:33]: I reworded to just say that the Bridge can be used to manage Keptn projects and services but I think this verbage can be improved.

File: content/docs/concepts/architecture/index.md (hunk 1 above)
Comment (StackScribe) [reply content/docs/concepts/architecture/index.md:69]: Good catch!    @thisthat I tried to do a little fix here based on the info in the 16.0.x Release Notes but it looks like the resource-service was in Keptn as a Beta feature from 0.12.x forward.   The only other mentions I find in the docs are in the reference page for keptn_add_resource and that isn't a lot of information.  Please advise.

File: content/docs/concepts/architecture/index.md (hunk 2 above)
Comment (StackScribe) [reply content/docs/concepts/architecture/index.md:105]: @thisthat Where could I find some info about this control plane connector?  And how to reword this?

Comment (agardnerIT) [reply content/docs/concepts/architecture/index.md:105]: My understanding in terms of "maturity, when they were introduced and functionality":

1) distributor (the original mechanism)
2) [cp-connector (control plane connector)](https://github.com/keptn/go-utils/tree/master#creating-a-keptn-service-using-cp-connector) handles the logic of an integration connecting back to the control plane. Benefit is that it's handled in Go code and the distributor pod is no longer required. Drawback - you're coding it. Currently used by all "core" services.
3) [go-sdk](https://github.com/keptn/go-utils/tree/master#creating-a-keptn-service-using-the-go-sdk-experimental). [experimental] A wrapper around the cp-connector which adds features.

For me, this is very deep technical knowledge and shouldn't really live in the main docs?

File: content/docs/concepts/architecture/index.md [hunk 5]
```diff
@@ -9,73 +9,137 @@ Keptn is an event-based control plane for continuous delivery and automated oper
...
 
 ## Keptn CLI
-The Keptn CLI needs to be installed on the local machine and is used to send commands to Keptn by interacting with the API of Keptn. To communicate with Keptn you need to know a shared secret that is generated during the installation and verified by the *api* component.
 
-**Note:** A dedicated Keptn CLI section is provided [here](../../0.15.x/reference/cli/), which helps you to get started and lists all available commands.
+Use the [Keptn CLI](../../0.18.x/reference/cli/) to send commands
+that interact with the [Keptn API](../../0.18.x/reference/api/).
+It must be [installed](../../install/cli-install)
+on the local machine and is used to send commands to Keptn.
+To communicate with Keptn, you need to know a shared secret
```
[the remark this answers is not in the document]
Comment (StackScribe) [reply content/docs/concepts/architecture/index.md:26]: I think it is but it is managed by the "secret service" so it's all a bit confusing.  I tried rewording this -- see if it works.

File: content/docs/concepts/architecture/index.md (hunk 2 above)
Comment (StackScribe) [reply content/docs/concepts/architecture/index.md:105]: I think this sort of information very much belongs in the architecture document for the product, at some level.  I just incorporated Adam's explanation, which could be expanded but it clarifies things a lot.  We talk about cp-connector and go-sdk on Slack and I hear lots of comments about the distributor eventually going away plus they have been mentioned in the Release Notes.  It seems misleading to only mention the distributor. 

See what you think of my rewrite.  The Architecture section needs some serious attention and I didn't really mean to do it in this PR.  This does incrementally improve the information that was there.

**Review by agardnerIT — APPROVED** [review]
All good except:

helm-service is no longer included in core from 0.18.2 onwards.

Other than that, it's good to go.

**Comment by StackScribe** [comment]
Aw shoot!  Also need to fix line 151 to say "distributor sidecars or cp-connector".  Stay tuned.

**Pull request merged**